sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/WebApplicationFirewallPolicyImpl.java

```diff
@@ -0,0 +1,252 @@
+        ensurePolicySettings().withRequestBodyCheck(false);
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withRequestBodySizeLimitInKb(int limitInKb) {
+        ensurePolicySettings()
+            .withRequestBodyInspectLimitInKB(limitInKb)
+            .withMaxRequestBodySizeInKb(limitInKb);
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withFileUploadSizeLimitInMb(int limitInMb) {
+        ensurePolicySettings()
+            .withFileUploadLimitInMb(limitInMb);
+        return this;
+    }
+
+    @Override
+    public void beforeGroupCreateOrUpdate() {
+        super.beforeGroupCreateOrUpdate();
+        ensureDefaultCoreRuleSet();
+    }
+
+    @Override
+    public Mono<WebApplicationFirewallPolicy> createResourceAsync() {
+        return this.manager().serviceClient().getWebApplicationFirewallPolicies()
+            .createOrUpdateAsync(this.resourceGroupName(), this.name(), this.innerModel())
+            .map(innerToFluentMap(this));
+    }
+
+    @Override
+    protected Mono<WebApplicationFirewallPolicyInner> getInnerAsync() {
+        return this.manager().serviceClient().getWebApplicationFirewallPolicies()
+            .getByResourceGroupAsync(this.resourceGroupName(), this.name());
+    }
+
+    private PolicySettings ensurePolicySettings() {
+        if (this.innerModel().policySettings() == null) {
+            PolicySettings policySettings = new PolicySettings();
+            this.innerModel().withPolicySettings(policySettings);
+        }
+        return this.innerModel().policySettings();
+    }
+
+    private ManagedRulesDefinition ensureManagedRules() {
+        ManagedRulesDefinition managedRulesDefinition = this.innerModel().managedRules();
+        if (managedRulesDefinition == null) {
+            managedRulesDefinition = new ManagedRulesDefinition();
+            this.innerModel().withManagedRules(managedRulesDefinition);
+            if (managedRulesDefinition.managedRuleSets() == null) {
+                managedRulesDefinition.withManagedRuleSets(new ArrayList<>());
+            }
+            if (managedRulesDefinition.exclusions() == null) {
+                managedRulesDefinition.withExclusions(new ArrayList<>());
+            }
+        }
+        return managedRulesDefinition;
+    }
+
+    private void ensureDefaultCoreRuleSet() {
+        ensureManagedRules();
+        if (this.innerModel().managedRules()
+            .managedRuleSets()
+            .stream()
+            .allMatch(ruleSet -> BOT_DETECTION_RULE_SET_TYPE.equals(ruleSet.ruleSetType()))) {
+            ManagedRuleSet defaultRuleSet =
+                new ManagedRuleSet()
+                    .withRuleSetType(DEFAULT_MANAGED_RULE_SET)
+                    .withRuleSetVersion(DEFAULT_MANAGED_RULE_SET_VERSION);
+            this.innerModel().managedRules().managedRuleSets().add(defaultRuleSet);
+        }
```

Comment:
some comment about this logic?

Why we check `allMatch` on `BOT_DETECTION_RULE_SET_TYPE`? Should the condition be that it has `BOT_DETECTION_RULE_SET_TYPE`, but not has `DEFAULT_MANAGED_RULE_SET`.

Not sure if future there be more types.

Reply:
There exist two types of mandatory managed rule sets for WAF policy for Application Gateway: OWASP(CRS) and Microsoft_DefaultRuleSet(DRS).
<img width="1586" alt="Screenshot 2023-06-14 at 4 14 36 PM" src="https://github.com/Azure/azure-sdk-for-java/assets/92354331/f8ec55d7-9983-4f58-b6d8-df9c5fec32bb">

Yes, it's better to check for CRS or DRS as they are less prone to change.

Or should we make managed rule set a mandatory stage and let user choose for their own(like CLI and Portal)? E.g.
```java
networkManager.webApplicationFirewallPolicies()
    .define(policyName)
    .withRegion(region)
    .withExistingResourceGroup(rgName)
    .withPopularManagedRuleSet(KnownManagedRuleSet.OWASP_3_2) // mandatory stage
    //.withManagedRuleSet(new ManagedRuleSet().withRuleSetType("OWASP").withRuleSetVersion("3.2"))
    .xxx()
    .create()
```